src/newTest/java/org/opensearch/test/framework/cluster/LocalOpenSearchCluster.java

```diff
@@ -0,0 +1,610 @@
+                    try {
+                        node.start();
+                        running = true;
+                        completableFuture.complete("initialized");
+                    } catch (BindTransportException | BindHttpException e) {
+                        log.warn("Port collision detected for {}", this, e);
+                        portCollision = true;
+                        try {
+                            node.close();
+                        } catch (IOException e1) {
+                            log.error(e1);
+                        }
+
+                        node = null;
+                        PortAllocator.TCP.reserve(transportPort, httpPort);
+
+                        completableFuture.complete("retry");
+
+                    } catch (Throwable e) {
+                        log.error("Unable to start {}", this, e);
+                        node = null;
+                        completableFuture.completeExceptionally(e);
+                    }
+                }
+            }).start();
+
+            return completableFuture;
+        }
+
+        public Client getInternalNodeClient() {
+            return node.client();
+        }
+
+        public PluginAwareNode esNode() {
+            return node;
+        }
+
+        public boolean isRunning() {
+            return running;
+        }
+
+        public <X> X getInjectable(Class<X> clazz) {
+            return node.injector().getInstance(clazz);
+        }
+
+        public void stop() {
+            try {
+                log.info("Stopping {}", this);
+
+                running = false;
+
+                if (node != null) {
+                    node.close();
+                    node = null;
+                    Thread.sleep(10);
+                }
+
+            } catch (Throwable e) {
+                log.warn("Error while stopping " + this, e);
+            }
+        }
+
+        @Override
+        public String toString() {
+            String state = running ? "RUNNING" : node != null ? "INITIALIZING" : "STOPPED";
+
+            return nodeName + " " + state + " [" + transportPort + ", " + httpPort + "]";
+        }
+
+        public boolean isPortCollision() {
+            return portCollision;
+        }
+
+        public String getNodeName() {
+            return nodeName;
+        }
+
+        @Override
+        public InetSocketAddress getHttpAddress() {
+            return httpAddress;
+        }
+
+        @Override
+        public InetSocketAddress getTransportAddress() {
+            return transportAddress;
+        }
+
+        private Settings getEsSettings() {
+            Settings settings = getMinimalEsSettings();
+
+            if (nodeSettingsSupplier != null) {
+                // TODO node number
+                return Settings.builder().put(settings).put(nodeSettingsSupplier.get(0)).build();
+            }
+
+            return settings;
+        }
+
+        private Settings getMinimalEsSettings() {
```

Comment:
Rename to getMinimalSettings

Reply:
What about reusing the minimumOpenSearchSettingsBuilder?

Reply:
In my opinion, we should not introduce any changes here. The code in the class `MinimumSecuritySettingsSupplierFactory` contains settings related to TLS/SSL, whereas the code in method `Node#getMinimalOpenSearchSettings` returns basic settings, like node name, working directory, etc. Therefore, I think that responsibilities should not be mixed here. @peternied  what is your opinion?